### PR TITLE
[Feature Fix] Fix  watch count to show impossible numbers

### PR DESCRIPTION
**Pull request by haoyuchen1992**
### Purpose
It makes it impossible to increase or decrease watch count multiple time in a row. 
Resolved the bug -- #3201

### Changes
Because the reason for the issue is that it is still open to increase the count when the update is in progress. So normally the change should be only made when there is no updating in progress. The problem's logic is similar as the #3203. 

1) Add status variable -- `watchUpdateInProgress`
2) Add conditional judge before updating watch counts
3) Update the status variable when the watch update is done or started.

### SideEffects
None.

### CODE REVIEW

**Comment by sloria** [comment]
While I think this approach works, it will still send a request for every click the user makes, which is not ideal.

I think the ideal solution is simpler: debounce the ajax call (we have `$osf.debounce`) so that button-mashing doesn't make a bunch of requests.

@haoyuchen1992 Does this make sense?

**Comment by haoyuchen1992** [comment]
@sloria Firstly, I tried to use `$osf.debounce`. However, there is a problem -- what wait time should I set? . The time have to be larger than the time for one request. But the time of ajax request is totally dependent on the internet.  If the time is too long, it will influence the responsiveness.  

Secondly, after checking again, my current code will not send a request for every click. It only sends the request when there is not request in progress. So it looks like the current solution is more accurate than using `$osf.debounce`.

File: website/static/js/nodeControl.js [hunk 1]
```diff
@@ -234,23 +234,31 @@ var ProjectViewModel = function(data) {
     /**
      * Toggle the watch status for this project.
      */
+    var watchUpdateInProgress = false;
+    var noWatchUpdateInProgress = function (){
```
Comment (sloria) [inline website/static/js/nodeControl.js]: Is this function necessary? I don't think it adds any more clarity than doing `!watchUpdateInProgress`.

File: website/static/js/nodeControl.js [hunk 2]
```diff
@@ -234,23 +234,31 @@ var ProjectViewModel = function(data) {
...
+            } else {
+                self.watchedCount(self.watchedCount() + 1);
+            }
+            watchUpdateInProgress = true;
+            osfHelpers.postJSON(
+                self.apiUrl + 'togglewatch/',
+                {}
+            ).done(function (data) {
+                    // Update watch count in DOM
+                    watchUpdateInProgress = false;
```
Comment (sloria) [inline website/static/js/nodeControl.js]: Fix indentation here.

**Comment by sloria** [comment]
OK, thanks for the clarification. After taking a closer look, the current approach works well. Just a few minor edits, and this is good to merge. :sunrise_over_mountains:

**Pull request merged**